fs/vfs/fs_fsync.c

```diff
@@ -55,13 +55,6 @@ int file_fsync(FAR struct file *filep)
 {
   struct inode *inode;
 
-  /* Was this file opened for write access? */
-
-  if ((filep->f_oflags & O_WROK) == 0)
```

Comment:
What is the point of `fsync` in case if file is not open for write? Maybe reasonable just return `OK` in this case instead of removing check? I mean that we really do not need to calll `inode->​u​.​i_mops​->​sync​` if file is not open for writing.

Reply:
eg. you can make modifications to the file using another open descriptor.

Reply:
Yes. I missed that. Thanks for clarification!